### ISSUE TITLE
Configuration to disable queries on non indexed payload field

### ISSUE THREAD

**Issue by agourlay** [qdrant/qdrant#2950]
**Is your feature request related to a problem? Please describe.**

Querying a non indexed payload field can become rapidly a problem at scale.
It is sometimes non trivial for an operator to realize this and to create an index on a production deployment.

Based on experience, it is often the case that the users actually overlooked the payload field and forgot to index it in first place.

It would be great to be able to configure a collection to not run any filter on non indexed payload field.

This would make deployments safer in an opt-in fashion especially if they can receive arbitrary queries.

**Describe the solution you'd like**

A new collection configuration field `disable_unindexed_payload_search` which will turn search queries targeting non indexed payload field into client errors.

**Comment by timvisee** [comment]
I like this idea! 

Would this also block matches which type is different than the index type?

**Comment by agourlay** [comment]
> Would this also block matches which type is different than the index type?

Yes, the spirit being only well formed indexed payload can be queried.

**Comment by generall** [comment]
I am not sure about the collection config, it is kinda bloated already. 
Maybe it would be better to log warnings if filter is used on un-indexed fields and query took longer than X millis?

**Comment by coszio** [comment]
I like the warning logs idea, but maybe it could be based on cardinality rather than timing?

**Comment by timvisee** [comment]
> but maybe it could be based on cardinality rather than timing?

In both cases you've to define magic values. I think it's easier to express this in time.

**Comment by generall** [comment]
> I like the warning logs idea, but maybe it could be based on cardinality rather than timing?

we can't compute cardinality for unindexed fields.
